packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/composite_datasource.rb

```diff
@@ -0,0 +1,76 @@
+module ForestAdminDatasourceCustomizer
+  class CompositeDatasource
+    def initialize
+      @datasources = []
+    end
+
+    def live_query_connections
+      @datasources.each_with_object({}) do |ds, acc|
+        acc.merge!(ds.live_query_connections || {})
+      end
+    end
+
+    def schema
+      { charts: @datasources.flat_map { |ds| ds.schema[:charts] } }
+    end
+
+    def collections
+      @datasources.reduce({}) { |acc, ds| acc.merge(ds.collections) }
+    end
+
+    def get_collection(name)
+      @datasources.each do |ds|
+        return ds.get_collection(name)
+      rescue StandardError
+        # ignore et continue
+      end
+
+      raise ForestAdminDatasourceToolkit::Exceptions::ForestException,
+            "Collection '#{name}' not found. List of available collections: #{collections.keys.sort.join(", ")}"
+    end
+
+    def render_chart(caller_obj, name)
+      @datasources.each do |ds|
+        return ds.render_chart(caller_obj, name) if ds.schema[:charts].include?(name)
+      end
+
+      raise ForestAdminAgent::Http::Exceptions::NotFoundError, "Chart '#{name}' is not defined in the dataSource."
```

Comment:
Here as well, I am not sure that a 404 is the agreed upon status, nodev2 seems to throw a basic Error